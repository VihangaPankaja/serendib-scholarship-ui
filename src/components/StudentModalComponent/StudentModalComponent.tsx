import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';


interface ModalPropType {
    show: boolean,
    studentId: string,
    onHide(): void
}


export default class StudentModalComponent extends Component<ModalPropType> {
    constructor(props: ModalPropType) {
        super(props);
    }

    render(): React.ReactNode {
        return <Modal {...this.props}>
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                    Details of Student {this.props.studentId}
                </Modal.Title>
            </Modal.Header>
            <Modal.Body>

            </Modal.Body>
            <Modal.Footer>
                <Button onClick={this.props.onHide}>Save</Button>
                <Button onClick={this.props.onHide}>Delete</Button>
                <Button onClick={this.props.onHide}>Close</Button>
            </Modal.Footer>
        </Modal>
    }
}