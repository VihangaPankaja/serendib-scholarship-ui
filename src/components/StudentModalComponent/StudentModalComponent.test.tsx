import { render } from "@testing-library/react";
import React from "react";
import { describe, it, expect } from "@jest/globals";
import StudentModalComponent from "./StudentModalComponent";

describe("StudentModalComponent", () => {
  it("renders", () => {
    const wrapper = render(<StudentModalComponent show={true} studentid={"1"} onHide={jest.fn()} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it("not renders", () => {
    const wrapper = render(<StudentModalComponent show={false} studentid={"1"} onHide={jest.fn()} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});