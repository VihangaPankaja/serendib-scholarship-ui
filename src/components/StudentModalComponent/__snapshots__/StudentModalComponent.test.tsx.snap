// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentModalComponent not renders 1`] = `<div />`;

exports[`StudentModalComponent renders 1`] = `<div />`;
